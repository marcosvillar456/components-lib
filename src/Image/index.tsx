import React from 'react';
import ImageInterface from './types';

const Image = ({
  src,
  alt,
  width = 24,
  height = 24,
  className,
}: ImageInterface) => (
  <React.Fragment>
    <img
      src={src}
      alt={alt}
      width={width}
      height={height}
      className={className}
    />
  </React.Fragment>
);

export default Image;
