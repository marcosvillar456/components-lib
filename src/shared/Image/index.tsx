import type ImageInterface from './types';

const Image = ({
  src,
  alt,
  width = 24,
  height = 24,
  className,
}: ImageInterface) => (
  <figure className={className}>
    <img
      src={src}
      alt={alt}
      width={width}
      height={height}
    />
  </figure>
);

export default Image;
