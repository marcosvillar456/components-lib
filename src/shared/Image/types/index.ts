type ImageInterface = {
  src: string;
  alt: string;
  width?: number;
  height?: number;
  className?: string;
}

export default ImageInterface;
