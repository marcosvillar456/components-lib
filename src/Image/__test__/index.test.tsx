import '@testing-library/jest-dom';
import { render, screen, cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';
import Image from '..';

describe('Image component', () => {
  afterEach(() => {
    jest.clearAllMocks();
    jest.resetAllMocks();
    jest.restoreAllMocks();
    cleanup();
  });

  const props = {
    src: 'https://example.com/image.jpg',
    alt: 'Test image',
    width: 100,
    height: 100,
    className: 'test-class',
  };

  test('snapshot of component', () => {
    const component = create(<Image {...props} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('renders an image with provided props', () => {
    render(<Image {...props} />);

    const imageElement = screen.getByAltText(props.alt);

    expect(imageElement).toBeInTheDocument();
    expect(imageElement).toHaveAttribute('src', props.src);
    expect(imageElement).toHaveAttribute('width', props.width.toString());
    expect(imageElement).toHaveAttribute('height', props.height.toString());
    expect(imageElement).toHaveClass(props.className);
  });
});
