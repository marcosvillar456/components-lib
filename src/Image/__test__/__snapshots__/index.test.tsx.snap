// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component snapshot of component 1`] = `
<img
  alt="Test image"
  className="test-class"
  height={100}
  src="https://example.com/image.jpg"
  width={100}
/>
`;
